src/main/java/org/apache/sysds/runtime/matrix/data/MatrixBlock.java

```diff
@@ -2730,7 +2730,23 @@ public MatrixBlock scalarOperations(ScalarOperator op, MatrixValue result) {
 		
 		return ret;
 	}
-
+	//ToDo
+	/*
+	extend the runtime operations in MatrixBlock
+	unaryOperations(UnaryOperator op, MatrixValue result),
+	which are called from UnaryMatrixCPInstruction,
+	so they exploit the update-in-place flag
+	by directly overwriting the input
+	(this can be done by simply assigning the input instead of allocating a new block).
+	So far, we only support this in the branch for cumaggregateUnaryMatrix,
+	but you should do that for multi-threaded and denseUnary/sparseUnary operations
+	on dense inputs&outputs (if both are dense) too
+	(later we'll extend that to sparse formats).
+	All extensions would be local to unaryOperations and potentially denseUnaryOperations
+	and sparseUnaryOperations.
+	Have a look into the cumaggregateUnaryMatrix to see how to use the flag from the passed
+	operator object.
+	 */
```

Comment:
Please remove this comment, instead do what you suggest, and throw a NotImplementedException in cases that are not implemented yet.